### PR TITLE
fix: logical errors in calculating monthly stats

### LINKED ISSUE #839: The monthly stat value is not accurate.
### What Operating System(s) are you seeing this problem on?

macOS (Apple Silicon)

### Backend.AI version

22.09.4

### Describe the bug

[`get_time_binned_monthly_stats()`](https://github.com/lablup/backend.ai/blob/main/src/ai/backend/manager/api/resource.py#L550) has a logical error, such as `num_sessions` not being counted properly even though the session is being used at that time. For example, if a user has used a session for the time period from 10:00 to 17:00, the resource quota should remain floating during that time. However, it is currently counted only at the session start time. It needs to modify it to calculate the proper stat.

### To Reproduce

_No response_

### Expected Behavior

Both the number and resources of the session must be properly combined during the session run time.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Sujin-Kim1**
this resolves #839 
Modify calculating monthly stats logic so that number of sessions and resource usage is calculated appropriately during all session execution times.

### CODE REVIEW

File: src/ai/backend/manager/api/resource.py [hunk 1]
```diff
@@ -601,29 +601,27 @@ async def get_time_binned_monthly_stats(request: web.Request, user_uuid=None):
...
+                if "cuda.shares" in row.occupied_slots:
+                    gpu_allocated += Decimal(row.occupied_slots["cuda.shares"])
+                raw_stat = await redis_helper.execute(
+                    root_ctx.redis_stat, lambda r: r.get(str(row["id"]))
+                )
+                if raw_stat:
+                    last_stat = msgpack.unpackb(raw_stat)
+                    io_read_bytes += int(nmget(last_stat, "io_read.current", 0))
+                    io_write_bytes += int(nmget(last_stat, "io_write.current", 0))
+                    disk_used += int(nmget(last_stat, "io_scratch_size/stats.max", 0, "/"))
```
Comment (fregataa) [inline src/ai/backend/manager/api/resource.py]:
```suggestion
        for row in rows:
            if ts + time_window <= row.created_at.timestamp() or ts >= row.terminated_at.timestamp():
                continue
            num_sessions += 1
            cpu_allocated += int(row.occupied_slots.get("cpu", 0))
            mem_allocated += int(row.occupied_slots.get("mem", 0))
            if "cuda.devices" in row.occupied_slots:
                gpu_allocated += int(row.occupied_slots["cuda.devices"])
            if "cuda.shares" in row.occupied_slots:
                gpu_allocated += Decimal(row.occupied_slots["cuda.shares"])
            raw_stat = await redis_helper.execute(
                root_ctx.redis_stat, lambda r: r.get(str(row["id"]))
            )
            if raw_stat:
                last_stat = msgpack.unpackb(raw_stat)
                io_read_bytes += int(nmget(last_stat, "io_read.current", 0))
                io_write_bytes += int(nmget(last_stat, "io_write.current", 0))
                disk_used += int(nmget(last_stat, "io_scratch_size/stats.max", 0, "/"))
```
We can simply iterate the rows.

**Review by fregataa — CHANGES_REQUESTED** [review]
Let's replace `while` to `for ... in` statement

File: src/ai/backend/manager/api/resource.py (hunk 1 above)
Comment (agatha197) [reply src/ai/backend/manager/api/resource.py]: reflected! thanks :)

**Review by fregataa — APPROVED** [review]
lgtm

**Pull request merged**